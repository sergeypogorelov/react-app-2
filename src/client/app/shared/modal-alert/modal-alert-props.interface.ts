export interface ModalAlertProps {
  title?: string;
  zIndex?: number;

  onClose(): void;
}
