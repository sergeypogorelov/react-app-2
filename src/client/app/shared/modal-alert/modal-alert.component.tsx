import React, { FunctionComponent } from 'react';

import { ModalAlertProps } from './modal-alert-props.interface';

export const DEF_Z_INDEX = 10000;

export const DEF_TITLE = 'Alert';

export const ModalAlert: FunctionComponent<ModalAlertProps> = ({
  children,
  title,
  zIndex,
  onClose,
}) => {
  const style: any = {
    zIndex: zIndex || DEF_Z_INDEX,
  };

  return (
    <div className="modal d-block" style={style}>
      <div className="modal-dialog modal-dialog-centered">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title">{title || DEF_TITLE}</h5>
          </div>
          <div className="modal-body">{children}</div>
          <div className="modal-footer justify-content-center">
            <button className="btn btn-primary" type="button" onClick={onClose}>
              OK
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
