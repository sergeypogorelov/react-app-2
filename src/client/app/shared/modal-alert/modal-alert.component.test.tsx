import React from 'react';
import renderer from 'react-test-renderer';

import { ModalAlertProps } from './modal-alert-props.interface';
import { ModalAlert } from './modal-alert.component';

function createTree(props?: Partial<ModalAlertProps>, children?: any) {
  if (!props) {
    props = {};
  }

  const defProps: ModalAlertProps = {
    onClose: jest.fn(),
  };

  const finalProps = {
    ...defProps,
    ...props,
  };

  return renderer.create(<ModalAlert {...finalProps}>{children}</ModalAlert>);
}

describe('ModalAlert component', () => {
  const testChildren = 'test';

  it('should render correctly without props and children', () => {
    const tree = createTree().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly without props and with children', () => {
    const tree = createTree(null, testChildren).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with title specified and children', () => {
    const tree = createTree({ title: 'test-title' }, testChildren).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with zIndex specified and children', () => {
    const tree = createTree({ zIndex: 101 }, testChildren).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
