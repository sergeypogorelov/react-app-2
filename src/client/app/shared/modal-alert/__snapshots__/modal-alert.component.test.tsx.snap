// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAlert component should render correctly with title specified and children 1`] = `
<div
  className="modal d-block"
  style={
    Object {
      "zIndex": 10000,
    }
  }
>
  <div
    className="modal-dialog modal-dialog-centered"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          test-title
        </h5>
      </div>
      <div
        className="modal-body"
      >
        test
      </div>
      <div
        className="modal-footer justify-content-center"
      >
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
          type="button"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalAlert component should render correctly with zIndex specified and children 1`] = `
<div
  className="modal d-block"
  style={
    Object {
      "zIndex": 101,
    }
  }
>
  <div
    className="modal-dialog modal-dialog-centered"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Alert
        </h5>
      </div>
      <div
        className="modal-body"
      >
        test
      </div>
      <div
        className="modal-footer justify-content-center"
      >
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
          type="button"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalAlert component should render correctly without props and children 1`] = `
<div
  className="modal d-block"
  style={
    Object {
      "zIndex": 10000,
    }
  }
>
  <div
    className="modal-dialog modal-dialog-centered"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Alert
        </h5>
      </div>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer justify-content-center"
      >
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
          type="button"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalAlert component should render correctly without props and with children 1`] = `
<div
  className="modal d-block"
  style={
    Object {
      "zIndex": 10000,
    }
  }
>
  <div
    className="modal-dialog modal-dialog-centered"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Alert
        </h5>
      </div>
      <div
        className="modal-body"
      >
        test
      </div>
      <div
        className="modal-footer justify-content-center"
      >
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
          type="button"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;
